src/Sylius/Behat/Element/Admin/ProductOption/FormElement.php

```diff
@@ -14,92 +14,57 @@
 namespace Sylius\Behat\Element\Admin\ProductOption;
 
 use Behat\Mink\Element\NodeElement;
-use Behat\Mink\Exception\ElementNotFoundException;
 use Sylius\Behat\Behaviour\ChecksCodeImmutability;
-use Sylius\Behat\Behaviour\NamesIt;
 use Sylius\Behat\Behaviour\SpecifiesItsField;
 use Sylius\Behat\Element\Admin\Crud\FormElement as BaseFormElement;
-use Webmozart\Assert\Assert;
 
 class FormElement extends BaseFormElement implements FormElementInterface
 {
-    use NamesIt;
     use SpecifiesItsField;
     use ChecksCodeImmutability;
 
-    public function addOptionValue(string $code, string $value): void
+    public function nameItIn(string $name, string $localeCode): void
     {
-        $count = count($this->getProductOptionValues());
-
-        $this->getElement('add_value')->click();
-        $this->getDocument()->waitFor(5, fn () => $count + 1 === count($this->getProductOptionValues()));
-
-        $optionValueForm = $this->getLastOptionValueElement();
-
-        $optionValueForm->fillField('Code', $code);
-
-        $optionValueForm->find('css', '[data-test-option-value-translation]')->setValue($value);
+        $this->getElement('name', ['%locale_code%' => $localeCode])->setValue($name);
     }
 
-    public function removeOptionValue(string $optionValue): void
+    public function addOptionValue(string $code, string $localeCode, string $value): void
     {
-        $count = count($this->getProductOptionValues());
-
-        $optionValues = $this->getProductOptionValues();
+        $this->getElement('add_value')->press();
+        $this->waitForFormUpdate();
 
-        foreach ($optionValues as $optionValueElement) {
-            if ($optionValueElement->has('css', sprintf('input[value="%s"]', $optionValue))) {
-                $optionValueElement->find('css', '[data-test-delete-value]')->click();
-
-                $this->getDocument()->waitFor(5, fn () => $count - 1 === count($this->getProductOptionValues()));
-
-                return;
-            }
-        }
-
-        throw new ElementNotFoundException($this->getSession(), 'option value', 'css', sprintf('input[value="%s"]', $optionValue));
+        $lastValue = $this->getElement('last_value');
+        $lastValue->fillField('Code', $code);
+        $lastValue->find('css', sprintf('[id$="_translations_%s_value"]', $localeCode))->setValue($value);
+        $this->waitForFormUpdate();
     }
 
-    public function nameItIn(string $name, string $localeCode): void
+    public function isThereOptionValue(string $optionValue, string $localeCode): bool
     {
-        $this->getElement('name', ['%localeCode%' => $localeCode])->setValue($name);
+        return $this->getElement('values')->has('css', sprintf('input[id$="_translations_%s_value"][value="%s"]', $localeCode, $optionValue));
```

Comment:
Can't we just extract the whole selector to defined elements and then just `hasElement`?